File: shortest job first/shortest_job_first.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
  int n,i,j,k;
  float total_wt=0.0,total_tat=0.0,ave_wt,ave_tat;
  cout<<"no of process:";
  cin>>n;
  int bt[n],wt[n],tat[n];
  cout<<"give burst time for each process:";
  for (j=0; j<n; j++)
  {
      cin>>bt[j];
  }
  sort(bt,bt+n);
  wt[0]=0;
  for (k=1;k<n;k++)
  {
      wt[k]=bt[k-1]+wt[k-1];
      total_wt=total_wt+wt[k];
  }
  for(i=0;i<n;i++)
  {
      tat[i]=bt[i]+wt[i];
      total_tat=total_tat+tat[i];
  }
ave_wt=total_wt/n;
ave_tat=total_tat/n;
cout<<"average waiting time: "<<ave_wt<<"\n";
cout<<"average waiting time: "<<ave_tat;
}

File: shortest job first/shortest job first.py
n=int(input("enter no of process:"))
list1=[]
for i in range(0,n):
    a=input("enter process:")
    b=int(input("burst time:"))
    list1.append([a,b])
list1.sort(key=lambda x:x[1]) # here (key}<--Specifies a function that determines the value used for comparison during sorting. and {lambda x: x[1]}<-- A small anonymous function that returns the second element of each item in the list.
wt=[]
wt.append(0)
for i in range(1,n):
    t=list1[i-1][1]+wt[i-1]
    wt.append(t)
total_wt=sum(wt)
print("average waiting time:",total_wt/n)
for i in range(0,n):
    print(f"process{list1[i][0]} started at {wt[i]} and end at { list1[i][1]+wt[i]}")
 